app/scripts/modules/core/pipeline/config/triggers/webhook/webhookTrigger.module.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.core.pipeline.trigger.webhook', [
+    require('../../../../../core/config/settings.js'),
+  ])
+  .config(function (pipelineConfigProvider) {
+    pipelineConfigProvider.registerTrigger({
+      label: 'Webhook',
+      description: 'Executes the pipeline on a Webhook POST',
+      key: 'webhook',
+      controller: 'WebhookTriggerCtrl as ctrl',
+      controllerAs: 'vm',
+      templateUrl: require('./webhookTrigger.html'),
+      popoverLabelUrl: require('./webhookPopoverLabel.html'),
+    });
+  })
+  .controller('WebhookTriggerCtrl', function (trigger, $scope) {
```

Comment:
If you add `settings` as an argument to the `function` here, you can include it in the controller, i.e.
```javascript
  .controller('WebhookTriggerCtrl', function (trigger, $scope, settings) {
    this.endpointUrl = [settings.gateUrl, 'webhooks', 'webhook', ''].join('/');
```
Then, in the HTML, you can reference it as:
```html
<div class="col-md-6">{{vm.endpointUrl}}</div>
```

That should get around the `localhost` issue mentioned above.